src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -661,14 +664,13 @@ public File createDataSetApprovedUsersDocument(Integer dataSetId) throws IOExcep
             for(Document dar: darList){
                 String referenceId = dar.getString(DarConstants.REFERENCE_ID);
                 DataAccessRequest dataAccessRequest = findByReferenceId(referenceId);
+                User user = userDAO.findUserById(dataAccessRequest.getUserId());
                 Date approvalDate = electionDAO.findApprovalAccessElectionDate(referenceId);
                 if (approvalDate != null) {
                     String email = userPropertyDAO
                             .findPropertyValueByPK(dataAccessRequest.getUserId(), DarConstants.ACADEMIC_BUSINESS_EMAIL);
-                    String name = userPropertyDAO
-                            .findPropertyValueByPK(dataAccessRequest.getUserId(), DarConstants.PROFILE_NAME);
-                    String institution = userPropertyDAO
-                            .findPropertyValueByPK(dataAccessRequest.getUserId(), DarConstants.INSTITUTION);
+                    String name = (user == null) ? "" : user.getDisplayName();
+                    String institution = (user == null || user.getInstitutionId() == null) ? "" : institutionDAO.findInstitutionById(user.getInstitutionId()).getName();
```

Comment:
Nice. 

This method is a great area for refactoring, as an FYI. Anywhere we see `Document`, we need to replace that with `DataAccessRequest` which is what a document really is. `Document` is how the DARs were stored when they were mongo objects and are just generic object holders. Moving to `DataAccessRequest` gives us a lot more control over the fields. 

If you dig into this method a little more, you'll see that we do this work horribly inefficiently:

1. get ALL DARs (see describeDataAccessByDataSetId)
2. convert DARs to documents (see describeDataAccessByDataSetId)
3. filter documents on those having dataset ids with the list (see describeDataAccessByDataSetId)
4. re-query for the DAR (see line 666)